app/assets/javascripts/discourse/app/models/post.js

```diff
@@ -568,10 +570,13 @@ export default class Post extends RestModel {
   }
 
   destroy(deletedBy, opts) {
+    this.set("isDeleting", true);
     return this.setDeletedState(deletedBy).then(() => {
```

Comment:
I think the finally might not be in the right spot? it only operates on the internal promise, we probably want outside. 

To simplify I would probably just make this an async function.